From 64a885842bb58913ac90f248ae96494d92fe184a Mon Sep 17 00:00:00 2001
From: Eric Chanudet <chanudete@ainfosec.com>
Date: Fri, 20 Jul 2018 17:52:17 -0400
Subject: [PATCH] i2c-hid: Silence "incomplete report" spam.

Attended once by:
ef6eaf27274c HID: i2c-hid: Fix "incomplete report" noise

Similar touchpad will spam:
[ 1538.596775] i2c_hid i2c-DLL07A0:01: i2c_hid_get_input: incomplete report (83/2)
... for each action on the touchpad and buttons.

Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>
---
 drivers/hid/i2c-hid/i2c-hid.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/hid/i2c-hid/i2c-hid.c b/drivers/hid/i2c-hid/i2c-hid.c
index d92827556389..3ead44decd39 100644
--- a/drivers/hid/i2c-hid/i2c-hid.c
+++ b/drivers/hid/i2c-hid/i2c-hid.c
@@ -477,8 +477,9 @@ static void i2c_hid_get_input(struct i2c_hid *ihid)
 	}
 
 	if ((ret_size > size) || (ret_size < 2)) {
-		dev_err(&ihid->client->dev, "%s: incomplete report (%d/%d)\n",
-			__func__, size, ret_size);
+		dev_err_once(&ihid->client->dev,
+				"%s: incomplete report (%d/%d)\n",
+				__func__, size, ret_size);
 		return;
 	}
 
-- 
2.17.0

